sstables_loader.cc

```diff
@@ -286,11 +347,20 @@ future<> sstable_streamer::stream_sstable_mutations(const dht::partition_range&
     co_return;
 }
 
+template <typename... Args>
+static std::unique_ptr<sstable_streamer> make_sstable_streamer(bool uses_tablets, Args&&... args) {
+    if (uses_tablets) {
+        return std::make_unique<tablet_sstable_streamer>(std::forward<Args>(args)...);
+    }
+    return std::make_unique<sstable_streamer>(std::forward<Args>(args)...);
+}
+
 future<> sstables_loader::load_and_stream(sstring ks_name, sstring cf_name,
         ::table_id table_id, std::vector<sstables::shared_sstable> sstables, bool primary_replica_only) {
-    sstable_streamer streamer(_messaging, _db.local(), table_id, std::move(sstables));
+    auto streamer = make_sstable_streamer(_db.local().find_column_family(table_id).uses_tablets(),
+                                          _messaging, _db.local(), table_id, std::move(sstables));
 
-    co_await streamer.stream(primary_replica_only);
+    co_await streamer->stream(primary_replica_only);
```

Comment:
What guarantees that the effective replication map lock is held around this?

Reply:
Sure.